pkg/route/api/validation/validation.go

```diff
@@ -125,23 +125,24 @@ func ExtendedValidateRoute(route *routeapi.Route) field.ErrorList {
 	//       break, so disable the hostname validation for now.
 	// hostname := route.Spec.Host
 	hostname := ""
-	var certPool *x509.CertPool
+	var verifyOptions *x509.VerifyOptions
 
 	if len(tlsConfig.CACertificate) > 0 {
-		certPool = x509.NewCertPool()
+		certPool := x509.NewCertPool()
 		if ok := certPool.AppendCertsFromPEM([]byte(tlsConfig.CACertificate)); !ok {
```

Comment:
AppendCertsFromPEM returns ok even if the CA bundle contains invalid certs, which could break the router. `pkg/cmd/util/CertificatesFromPEM()` will return an error if there are malformed certs, which is probably what you want.